open_spiel/algorithms/tabular_sarsa.cc

```diff
@@ -163,7 +163,30 @@ void TabularSarsaSolver::RunIteration() {
     double new_q_value = reward + discount_factor_ * next_q_value;
 
     double prev_q_val = values_[{key, curr_action}];
-    values_[{key, curr_action}] += learning_rate_ * (new_q_value - prev_q_val);
+    if (lambda_ == 0) {
+      // If lambda_ is equal to zero, run sarsa as usual. It's not necessary
+      // to update eligibility traces.
+      values_[{key, curr_action}] +=
+          learning_rate_ * (new_q_value - prev_q_val);
+    } else {
+      double lambda =
+          player != next_state->CurrentPlayer() ? -lambda_ : lambda_;
+      eligibility_traces_[{key, curr_action}] += 1;
+      std::string state;
+      Action action;
+      double prev_q_val_tmp;
+
+      for (auto q_cell : values_) {
+        state = q_cell.first.first;
+        action = q_cell.first.second;
+        prev_q_val_tmp = q_cell.second;
```

Comment:
Hi Giovanni, thanks for the changes.
Quick question regarding the prev_q_val_tmp variable as it is unused. 
Could this variable just be removed or should it actually have been used in the update below?

Reply:
Hi.
I missed that. It was there for debugging purposes, and I forgot to delete it.
I removed the variable.